pandas/tests/tseries/offsets/test_dst.py

```diff
@@ -173,3 +175,49 @@ def test_all_offset_classes(self, tup):
         first = Timestamp(test_values[0], tz="US/Eastern") + offset()
         second = Timestamp(test_values[1], tz="US/Eastern")
         assert first == second
+
+
+@pytest.mark.parametrize(
+    "case",
+    [
+        (
+            Timestamp("1900-01-01"),
+            Timestamp("1905-07-01"),
+            MonthBegin(66),
+            "Africa/Kinshasa",
+        ),  # GH41906
+        (
+            Timestamp("2021-10-01 01:15"),
+            Timestamp("2021-10-31 01:15"),
+            MonthEnd(1),
+            "Europe/London",
+        ),
+        (
+            Timestamp("2010-12-05 02:59"),
+            Timestamp("2010-10-31 02:59"),
+            SemiMonthEnd(-3),
+            "Europe/Paris",
+        ),
+        (
+            Timestamp("2021-10-31 01:20"),
+            Timestamp("2021-11-07 01:20"),
+            CustomBusinessDay(2, weekmask="Sun Mon"),
+            "US/Eastern",
+        ),
+        (
+            Timestamp("2020-04-03 01:30"),
+            Timestamp("2020-11-01 01:30"),
+            YearBegin(1, month=11),
+            "America/Chicago",
+        ),
+    ],
+)
+def test_nontick_offset_with_ambiguous_time_error(case):
+    # .apply for non-Tick offsets throws AmbiguousTimeError when the target dt
+    # is dst-ambiguous
+    original_dt, target_dt, offset, tz = case
```

Comment:
Could you make these parameterized arguments instead? i.e.

`def test_nontick_offset_with_ambiguous_time_error(original_dt, target_dt, offset, tz):`

Reply:
no problem